Lib/test/test_csv.py

```diff
@@ -736,6 +736,36 @@ def test_write_field_not_in_field_names_ignore(self):
         csv.DictWriter.writerow(writer, dictrow)
         self.assertEqual(fileobj.getvalue(), "1,2\r\n")
 
+    def test_dict_reader_accepts_only_sequences(self):
```

Comment:
The test name is too sweeping - you check that iterator of a list doesn't work and that list does work, that not proof that "only sequences" are accepted.